runtime/interpreter/interpreter.go

```diff
@@ -582,7 +582,9 @@ func (interpreter *Interpreter) Interpret() (err error) {
 		err = internalErr
 	})
 
-	interpreter.Program.Program.Accept(interpreter)
+	if interpreter.Program != nil {
```

Comment:
do we need a condition on interpreter.Program.Program != nil as well?

Reply:
No, if a program is provided, the AST and Elaboration are required. 

Unfortunately Go doesn't allow indicating if something is optional or not